chapter05/red/exercise2.pl

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+use List::Util qw( reduce );
+
+my $fn = shift @ARGV // 'coconet.dat';
+open my $fh, '<', $fn;
+
+my %stats;
+while( <$fh> ) {
+  next if /^\s*(#.*)?$/;
```

Comment:
i don't think you really needed to create a pattern that matches the whole line. 

I'm not entirey sure how perl handles this but in some languages i don't think this is a valid pattern, as it can match nothing. ie.  __0__ or more spaces, and then __0__ or 1 hash followed by any character. My regex test website doesn't treat `/^$/` as a valid pattern.